### PR TITLE
Dns

### PR DESCRIPTION
**Pull request by lkolluru05**

### CODE REVIEW

File: chaosgcp/__init__.py [hunk 1]
```diff
@@ -331,4 +331,5 @@ def load_exported_activities() -> List[DiscoveredActivities]:
     activities.extend(discover_actions("chaosgcp.lb.actions"))
     activities.extend(discover_probes("chaosgcp.neg.probes"))
     activities.extend(discover_actions("chaosgcp.neg.actions"))
+    activities.extend(discover_actions("chaosgcp.dns.actions"))
```
Comment (Lawouach) [inline chaosgcp/__init__.py:334]: Good catch

File: chaosgcp/dns/actions.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+__all__ = ["update_service"]
+
+
+def update_service(
+    secrets: Secrets = None,
+    project: str = None,
+    managed_zone: str = None,
+    ip_address: str="10.0.0.6",
+    name: str="b2e453371f40.34gubvy95ghwe.us-central1.sql.goog.",
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: probably not a good idea to leave this default value

File: chaosgcp/dns/actions.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    zone_name: str="pscsqlzone"):
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: likewise

File: chaosgcp/dns/actions.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+from chaosgcp import load_credentials
+
+__all__ = ["update_service"]
+
+
+def update_service(
+    secrets: Secrets = None,
+    project: str = None,
+    managed_zone: str = None,
+    ip_address: str="10.0.0.6",
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: missing spaces around `=`

Comment (Lawouach) [reply chaosgcp/dns/actions.py]: also remove that default value

File: chaosgcp/dns/actions.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+from chaoslib.types import Configuration, Secrets
+from googleapiclient import discovery
+
+from chaosgcp import load_credentials
+
+__all__ = ["update_service"]
+
+
+def update_service(
+    secrets: Secrets = None,
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py:20]: this should always be the last parameter (as a convention)

File: chaosgcp/dns/actions.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    project: str = None,
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: that should be coming from the configuration

**Review by Lawouach — CHANGES_REQUESTED** [review]
I've made a few comments. We'll go through your changes as you amend these.

Please also add a changelog entry (don't set the version) as well as tests.

File: chaosgcp/dns/actions.py (hunk 6 above)
Comment (Lawouach) [reply chaosgcp/dns/actions.py]: see https://github.com/chaostoolkit-incubator/chaostoolkit-google-cloud-platform/blob/master/chaosgcp/artifact/probes.py#L29

File: chaosgcp/dns/actions.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from typing import Any, Dict, List
+
+from chaoslib.types import Configuration, Secrets
+from googleapiclient import discovery
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: change this to

`from chaosgcp import client`

File: chaosgcp/dns/actions.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    name: str= None,
+    zone_name: str= None,
+    kind: str= "dns#resourceRecordSet",
+    ttl: int=5,
+    record_type: str="A",
+    existing_type: str="A"):
+
+    credentials = load_credentials(secrets)
+
+    service = discovery.build('dns', 'v1', credentials=credentials)
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: service = client('dns', 'v1', credentials=credentials)

**Review by Lawouach — CHANGES_REQUESTED** [review]

File: chaosgcp/dns/actions.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+from chaoslib.types import Configuration, Secrets
+from googleapiclient import discovery
+
+from chaosgcp import load_credentials
+
+__all__ = ["update_service"]
+
+
+def update_service(
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: the name of the action seems odd for a dns action, is that correct?

File: chaosgcp/dns/actions.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+from typing import Any, Dict
+
+from chaoslib.types import Configuration, Secrets
+
+from chaosgcp import client
+
+__all__ = ["update_service"]
+
+
+def update_service(
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: Please rename this action as it doesn't have a clear purpose

File: chaosgcp/dns/actions.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    ip_address: str = None,
+    name: str = None,
+    zone_name: str = None,
+    kind: str = "dns#resourceRecordSet",
+    ttl: int = 5,
+    record_type: str = "A",
+    existing_type: str = "A",
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+) -> Dict[str, Any]:
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py:21]: Please add a docstring.

File: chaosgcp/dns/actions.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+def update_service(
+    project: str = None,
+    ip_address: str = None,
+    name: str = None,
+    zone_name: str = None,
+    kind: str = "dns#resourceRecordSet",
+    ttl: int = 5,
+    record_type: str = "A",
+    existing_type: str = "A",
```
Comment (Lawouach) [inline chaosgcp/dns/actions.py]: Why are all these optionals? I can understand for some of them. But surely `name`, `zone_name` and `ip_address` should be mandatory?

File: chaosgcp/dns/actions.py (hunk 10 above)
Comment (Lawouach) [reply chaosgcp/dns/actions.py]: In fact, should we have more specialized actions like `add_A_record`, `set_cname`...

**Review by Lawouach — CHANGES_REQUESTED** [review]
It's a good start but it would be better to have more specialized actions with clear names and purpose than a generic function I think. It would make easier to set a clear goal in the experiment.

Also, please add a changelog entry (don't set the version at the top of the file).

Please also make sure to rebase from upstream your fork repository. It may not be up to date.

File: chaosgcp/dns/actions.py (hunk 10 above)
Comment (lkolluru05) [reply chaosgcp/dns/actions.py]: yes I can write them later

**Comment by Lawouach** [comment]
It seems you need to delint the code a bit @lkolluru05 :)

Consider also adding the changelog entry as well.

**Pull request closed without merging**